docs/source/topics/backends/snmp.rst

```diff
@@ -0,0 +1,149 @@
+************
+
+SNMP
+====
+
+SNMP (Simple Network Management Protocol) is a network protocol very useful for retrieving info from a device.
+All the information is retrieved by using codes called MIBs. All MIBs have a tree-like structure, every main information is the root and by adding more detail to the info
+the tree gains more depth.
+Obviously, by getting the smallest MIB which is "1" or simply " . " one can get all the tree.
+
+The base SNMP backend contains the following methods (some internal methods are not documented and are subject to change in the future):
+
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_dict**  | Returns a dict containing monitoring information depending on the type of the device.                                                    |
+|              | It follows the `NetJSON Devicemonitoring <https://netjson.org/rfc.html#name-devicemonitoring>`_ spec                                     |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **to_json**  | Calls the `to_dict` method and returns a JSON string of the dict                                                                         |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+| **validate** | Checks if connection with the device is working and raises `NetengineError` in case something is wrong                                   |
++--------------+------------------------------------------------------------------------------------------------------------------------------------------+
+
+Initializing an SNMP backend class requires the following arguments:
+
++---------------+---------------------------------------------------------------------+
+| **host**      | Management ip or hostname of the device                             |
++---------------+---------------------------------------------------------------------+
+| **community** | Community string for the SNMP connection. Default value is 'public' |
++---------------+---------------------------------------------------------------------+
+| **agent**     | Agent string for the SNMP connection                                |
++---------------+---------------------------------------------------------------------+
+| **port**      | Port for the SNMP connection. Default value is `161`                |
++---------------+---------------------------------------------------------------------+
+
+The SNMP backend provides support for 2 firmwares:
+ * AirOS
+ * OpenWRT
+
+.. note::
+
+    The data collected by Netengine is dependant on the OIDs available on your device. Some proprietary manufacturers may not
+    provide the same information as others.
+
+*****
+AirOS
+*****
+
```

Comment:
as in netjsonconfig, I suggest to make this a dedicated page, which should point to the base snmp page for the general concepts.

Some information regarding methods can be duplicated (no problem with this if we use autodoc).